### PR TITLE
Add script for license scanner

### PR DESCRIPTION
**Pull request by brianherrera**
This script will scan the source tree for license files and generate a file with the contents of all the licenses. This will be used by the installer team to generate the file for each release. This initial version performs a simple scan with filename patterns, there's upcoming work for additional filtering and utilizing packageinfo.json files. 

Tested in a workspace using cmake config to also include required 3rdParty packages. Test output: [NOTICES_TEST.txt](https://raw.githubusercontent.com/brianherrera/o3de/license_scanner/scripts/license_scanner/NOTICES_TEST.txt) (includes licenses in o3de and 3P packages)

Signed-off-by: brianherrera <briher@amazon.com>

### CODE REVIEW

**Review by amzn-changml — APPROVED** [review]
This works as a 1st pass (thanks for jumping on this, though). Subsequent passes should look for the file `Packageinfo.json` and retrieve the license through the path defined in there. Some 3p have multiple licenses, and what's in that json file will determine which one we selected. Getting all of them is fine, but obviously adds a bunch of extra text.

File: scripts/license_scanner/license_scanner.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                f.write(license_output)
+        return None
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Script to run LicenseScanner and generate license file')
+    parser.add_argument('--config-file', '-c', help='Config file for LicenseScanner')
+    parser.add_argument('--license-file-path', '-l', help='Create license file in the provided path')
+    parser.add_argument('--scan-path', '-s', default=os.curdir, help='Path to scan')
```
Comment (lumberyard-employee-dm) [inline scripts/license_scanner/license_scanner.py:110]: The default value is missing the parenthesis to invoke the method call. It should `os.curdir()`

Comment (lumberyard-employee-dm) [reply scripts/license_scanner/license_scanner.py:110]: Also all three of these parameters should use the type of `pathlib.Path` to enforce a path type is being supplied

File: scripts/license_scanner/license_scanner.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    try:
+        args = parse_args()
+        ls = LicenseScanner(args.config_file)
+        licenses = ls.scan(args.scan_path)
+
+        if args.license_file_path:
+            ls.create_license_file(licenses, args.license_file_path)
+    except Exception as e:
+        print(f'Type: {type(e).__name__}, Error: {e}')
+        return 1
```
Comment (lumberyard-employee-dm) [inline scripts/license_scanner/license_scanner.py:124]: It is not a good practice to globally catch an exceptions that are known about it.
Every python method that is invoked in this file has a specific set of exceptions that can occur and therefore the exception handling should be constrained to that level.

File: scripts/license_scanner/license_scanner.py (hunk 1 above)
Comment (brianherrera) [reply scripts/license_scanner/license_scanner.py:110]: Good call. I'll update the type for the parameters. 

os.curdir is the const string provided the OS for the cwd. You may be thinking of os.getcwd()

Comment (lumberyard-employee-dm) [reply scripts/license_scanner/license_scanner.py:110]: Sorry about that, my mistake.

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by AMZN-ScottR — APPROVED** [review]
Looks good to me.  The packaging script (at cmake/Packaging.cmake) will need to be updated to reflect the file name change as well.

**Review by amzn-changml — APPROVED** [review]

**Pull request merged**